### PR TITLE
Keep write context live until inverter thread has completed inver…

### PR DESCRIPTION
**Pull request by toregge**
…t task.

@geirst : please review
@baldersheim : FYI

### CODE REVIEW

File: searchcorespi/src/vespa/searchcorespi/index/iindexmanager.h [hunk 1]
```diff
@@ -94,7 +94,7 @@ class IIndexManager {
      * @param serialNum       The unique monotoninc increasing serial number
      *                        for this operation.
      **/
-    virtual void putDocument(uint32_t lid, const Document &doc, SerialNum serialNum) = 0;
+    virtual void putDocument(uint32_t lid, const Document &doc, SerialNum serialNum, OnWriteDoneType on_write_done) = 0;
```
Comment (geirst) [inline searchcorespi/src/vespa/searchcorespi/index/iindexmanager.h:97]: Please extend the function doc to include the new param.

File: searchcorespi/src/vespa/searchcorespi/index/imemoryindex.h [hunk 2]
```diff
@@ -43,7 +43,7 @@ struct IMemoryIndex : public searchcorespi::IndexSearchable {
      * @param lid the local document id.
      * @param doc the document to insert.
      */
-    virtual void insertDocument(uint32_t lid, const document::Document &doc) = 0;
+    virtual void insertDocument(uint32_t lid, const document::Document &doc, OnWriteDoneType on_write_done) = 0;
```
Comment (geirst) [inline searchcorespi/src/vespa/searchcorespi/index/imemoryindex.h:46]: Please extend the function doc to include the new param.

File: searchlib/src/vespa/searchlib/memoryindex/document_inverter.h [hunk 3]
```diff
@@ -76,7 +75,7 @@ class DocumentInverter {
      * For each text and uri field in the document a task for inverting and adding that
      * field (using a field inverter) is added to the 'invert threads' executor, then this function returns.
      **/
-    void invertDocument(uint32_t docId, const document::Document &doc);
+    void invertDocument(uint32_t docId, const document::Document &doc, OnWriteDoneType on_write_done);
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/memoryindex/document_inverter.h:78]: Unrelated to this change, but we should take a pass at the documentation of this class to ensure it is up to date.

**Review by geirst — APPROVED** [review]
👍

File: searchcorespi/src/vespa/searchcorespi/index/imemoryindex.h (hunk 2 above)
Comment (toregge) [reply searchcorespi/src/vespa/searchcorespi/index/imemoryindex.h:47]: Done.

File: searchcorespi/src/vespa/searchcorespi/index/iindexmanager.h (hunk 1 above)
Comment (toregge) [reply searchcorespi/src/vespa/searchcorespi/index/iindexmanager.h:100]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**